src/dsp_tools/utils/project_validate.py

```diff
@@ -13,6 +13,102 @@
 from dsp_tools.utils.excel_to_json_lists import expand_lists_from_excel
 
 
+def  check_for_undefined_superproperty(project_definition: dict[str, Any]) -> bool:
+    """
+    Check if the properties referenced in "super" of each property actually point to a property 
+    (DSP base properties and properties from other ontologies are not considered.)
+
+    Args:
+        project_definition: parsed JSON project definition
+
+    Raises:
+        BaseError: detailed error message
+
+    Returns:
+        True if the superproperties are valid
+    """
+    errors: dict[str, list[str]] = dict()
+    for onto in project_definition["project"]["ontologies"]:
+        ontoname = onto["name"]
+        propnames = [p["name"] for p in onto["properties"]]
+        for prop in onto["properties"]:
+            supers = prop["super"]
+            # form of supers:
+            #  - isSequenceOf  # DSP base property
+            #  - other:prop    # other onto
+            #  - same:prop     # same onto
+            #  - :prop         # same onto (short form)
+            
+            # filter out DSP base properties
+            supers = [s for s in supers if ":" in s]
+            # extend short form
+            supers = [regex.sub(r"^:", f"{ontoname}:", s) for s in supers]
+            # filter out other ontos
+            supers = [s for s in supers if regex.search(f"^{ontoname}:", s)]
+            # convert to short form
+            supers = [regex.sub(ontoname, "", s) for s in supers]
+
+            invalid_references = [s for s in supers if regex.sub(":", "", s) not in propnames]
+            if invalid_references:
+                errors[f"Ontology '{ontoname}', property '{prop['name']}'"] = invalid_references
+    
+    if errors:
+        err_msg = "Your data model contains properties that are derived from an invalid super-property:"
+        for loc, invalids in errors.items():
+            err_msg = err_msg + f"\n - {loc}: {invalids}"
+        raise BaseError(err_msg)
+    else:
+        return True
+
+
+def check_for_undefined_cardinalities(project_definition: dict[str, Any]) -> bool:
+    """
+    Check if the propnames that are used in the cardinalities of each resource are defined in the "properties" 
+    section. (DSP base properties and properties from other ontologies are not considered.)
```

Comment:
```suggestion
    section. (knora-base properties and properties from other ontologies are not considered.)
```

`knora-base` still exists as ontology. So I would call it like that.

Reply:
They are called like this
 - here: https://docs.dasch.swiss/2023.02.01/DSP-TOOLS/file-formats/json-project/caveats/#dsp-base-resources-and-base-properties-to-be-used-directly-in-the-xml-file
 - and here: https://docs.dasch.swiss/2023.02.01/DSP-TOOLS/file-formats/xml-data-file/#dsp-base-resources-and-base-properties-to-be-used-directly-in-the-xml-file

I think it's user friendlier to say DSP base property. Otherwise, I would have to rename the headings in the docs.